### ISSUE TITLE
enhancement(linter) new rule proposal/expand no useless spread

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#4921]
within some codebases, sometimes code such as:
```ts
const response: MyType = { foo: "bar" } 

return { ...response }
```

this normally occurs as a part of refactring (or perhaps logging the return value of a function.

however, the spread, in this case `{ ...response }` is a useless less code, allocates new memory and hurts performance.

Anytime the variable (in this case `response` is also used elsewhere, we would not report an error.

what do you all think? is this worth having?

**Comment by DonIsaac** [comment]
We could add this to `no-useless-spread` when we can determine if `response` is created as a literal. Otherwise, it's hard to tell if the programmer intentionally wants to clone here. We could have a new rule that checks for this in these cases.

**Comment by camc314** [comment]
yes. i think we should only report if it's a literal

we should not report cases such as:
```ts
const response: MyType = getResponse()

return { ...response }
```

when it's a literal (object or array), it's clear that there is a clone

**Comment by Boshen** [comment]
Feel free to implement this.

**Comment by Boshen** [comment]
Close as stale, unless you want to continue working on this.
